src/traccuracy/_tracking_graph.py

```diff
@@ -145,7 +145,17 @@ def __init__(
             name (str, optional): User specified name that will be included in result
                 outputs associated with this object
         """
-        self.segmentation = segmentation
+
+        # if the segmentation is not integer dtype, cast to uint64. Otherwise, leave it
+        # the original dtype
+        if segmentation is not None and segmentation.dtype.kind not in ["i", "u"]:
+            logger.warning(
```

Comment:
I think this should be a `UserWarning` not just a logger warning, because if the given arrays actually contain non-integral floats, they'll be truncated. I also think in that case, numpy will copy the array, so this could lead to memory issues for large segmentations as well.

Reply:
A) I didn't know about UserWarnings 😆 so thanks! B) I think in the end we should do the casting in the loaders (so that the seg_ids will match the casted data), and the tracking_graph can fail if the seg isn't integers. Two questions: 1. Do you agree that that approach makes sense? 2. Is the implmentation of the ctc loader that casts the data type if necessary as efficient as it can be? We might in-memory copy individual frames but not the whole dataset.

Reply:
I think doing it in the loaders makes more sense, yeah. I think as a user I'd be surprised to see the casting happen here. I think the general approaching of casting floats is ok, as long as we tell the user about it with a `UserWarning`. Maybe also will be good to add to the warning that if you actually have non-integral values, this will result in a copy.